src/mrinufft/operators/base.py

```diff
@@ -275,6 +278,55 @@ def __init_subclass__(cls):
         if backend := getattr(cls, "backend", None):
             cls.interfaces[backend] = (available, cls)
 
+    def check_shape_op(self, image, ksp=None):
```

Comment:
Why give ksp when you dont test its shape anyways?  We need to ideally check kspace's shape here if it is not None. I would recommend having the same function for op and adj_op